Add plugin: Day One Importer

MarcDonald opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

<!--- Paste a link to your repo here for easy access -->
Link to my plugin: https://github.com/marcdonald/obsidian-day-one-importer

## Release Checklist
- [x] I have tested the plugin on
  - [ ]  Windows
  - [x]  macOS
  - [ ]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files
  - [x] `main.js`
  - [x] `manifest.json`
  - [ ] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

MarcDonald commented:
/skip All violations are just in unit tests

joethei commented:
["description": "Allows you to import Day One JSON into Obsidian",](https://github.com/MarcDonald/obsidian-day-one-importer/blob/683c20ef20f1311719a0b7d9c6fcc73d6c1322bc/manifest.json#L6)
Don't include "Obsidian" in the description, also the description should be an action statement, like "Import Day One journals."

MarcDonald commented:
> @joethei ["description": "Allows you to import Day One JSON into Obsidian",](https://github.com/MarcDonald/obsidian-day-one-importer/blob/683c20ef20f1311719a0b7d9c6fcc73d6c1322bc/manifest.json#L6) Don't include "Obsidian" in the description, also the description should be an action statement, like "Import Day One journals."

Updated in version 0.5.2, I've also updated the minimum Obsidian version to 1.5.0 in case any of the APIs I use are newer.
